modules/azure/availabilityset.go

```diff
@@ -0,0 +1,132 @@
+package azure
+
+import (
+	"context"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/gruntwork-io/terratest/modules/collections"
+	"github.com/gruntwork-io/terratest/modules/testing"
+	"github.com/stretchr/testify/require"
+)
+
+// AvailabilitySetExists indicates whether the speficied Azure Availability Set exists
+func AvailabilitySetExists(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) bool {
+	exists, err := AvailabilitySetExistsE(t, avsName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return exists
+}
+
+// AvailabilitySetExistsE indicates whether the speficied Azure Availability Set exists
+func AvailabilitySetExistsE(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) (bool, error) {
+	_, err := GetAvailabilitySetE(t, avsName, resGroupName, subscriptionID)
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// CheckAvailabilitySetContainsVM checks if the Virtual Machine is contained in the Availability Set VMs
+func CheckAvailabilitySetContainsVM(t testing.TestingT, vmName string, avsName string, resGroupName string, subscriptionID string) bool {
+	avsVMs, err := GetAvailabilitySetVMsE(t, avsName, resGroupName, subscriptionID)
+	if err != nil {
+		return false
+	}
+
+	return collections.ListContains(avsVMs, strings.ToLower(vmName))
+}
+
+// GetAvailabilitySetVMs gets a list of VM names in the specified Azure Availability Set
+func GetAvailabilitySetVMs(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) []string {
+	vms, err := GetAvailabilitySetVMsE(t, avsName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return vms
+}
+
+// GetAvailabilitySetVMsE gets a list of VM names in the specified Azure Availability Set
+func GetAvailabilitySetVMsE(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) ([]string, error) {
+	client, err := GetAvailabilitySetClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	avs, err := client.Get(context.Background(), resGroupName, avsName)
+	if err != nil {
+		return nil, err
+	}
+
+	vms := []string{}
+
+	for _, vm := range *avs.VirtualMachines {
+		vms = append(vms, strings.ToLower(GetNameFromResourceID(*vm.ID)))
```

Comment:
Same question here on `strings.ToLower`.

Also, is a `nil` check necessary on `*vm.ID`?

Reply:
This is not necessary due to the for loop, if there are no VMs present, there will be no vm to nil panic on.